app/models/assignment.rb

```diff
@@ -42,6 +42,11 @@ class Assignment < ActiveRecord::Base
   # Because of app/views/main/_grade_distribution_graph.html.erb:25
   validates_presence_of :assignment_stat
 
+  # Assignments can now refer to themselves, where this is null if there
+  # is no parent (the same holds for the child peer reviews)
+  belongs_to :parent_assignment, class_name: 'Assignment', inverse_of: :peer_review
+  has_one :peer_review, class_name: 'Assignment', foreign_key: :parent_assignment_id, inverse_of: :parent_assignment
```

Comment:
Sorry, I meant to make a comment on this last time. I don't think "peer_review" is a good name for this association, as it conflicts with the other new model you've added. Maybe just refer to these as `pr_assignment` to keep it short.